packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -148,16 +148,19 @@ class AOTSnapshotter {
     // TODO(cbracken): replace IOSArch with TargetPlatform.ios_{armv7,arm64}.
     assert(platform != TargetPlatform.ios || iosArch != null);
 
-    final bool compileToSharedLibrary = preferSharedLibrary && androidSdk.ndkCompiler != null;
-    if (preferSharedLibrary && !compileToSharedLibrary) {
-      printStatus('Could not find NDK compiler. Not building in shared library mode.');
+    // buildSharedLibrary is Android-only.
+    assert(platform != TargetPlatform.ios || !buildSharedLibrary);
+
+    if (buildSharedLibrary && androidSdk.ndkCompiler == null) {
+      printError('Could not find NDK compiler. Unable to build with --build-shared-library');
+      return -2;
     }
 
     final PackageMap packageMap = new PackageMap(packagesPath);
     final String packageMapError = packageMap.checkValid();
     if (packageMapError != null) {
       printError(packageMapError);
-      return -2;
+      return -3;
```

Comment:
What are these magic numbers? Should they have a named constant declared somewhere for them?

Reply:
We should probably just replace them with a proper failure mode. They were initially there for our own use to differentiate the different types of failure. Those should be evident with proper trace logging either way.

Reply:
For now just replaced these with a uniform failed exit code. We should probably just move to throwToolExit like everywhere else since we never really use the exit codes other than to log error messages. Will do that in a followup.